src/pages/ArticlePage.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getArticles, getComments } from '../utils/api-requests.js';
+import SelectedArticle from '../components/SelectedArticle.jsx';
+import ArticleComments from '../components/ArticleComments.jsx';
+import Loading from '../components/Loading.jsx';
+
+const ArticlePage = () => {
+    const [selectedArticle, setSelectedArticle] = useState({});
+    const [articleComments, setArticleComments] = useState([]);
+    const [isLoadingArticle, setIsLoadingArticle] = useState(false);
+    const [isLoadingComments, setIsLoadingComments] = useState(false);
+    const { article_id } = useParams();
+
+    useEffect(() =>{
+        setIsLoadingArticle(true)
+        setIsLoadingComments(true)
+
+        getArticles(article_id)
```

Comment:
It could make sense to fetch the data here in different components maybe with the comments as a child of MainArticle?. If for instance the comments came really fast and the article took ages to come would it make sense to just show a list of comments?